afs-ext-base/src/main/java/com/powsybl/afs/ext/base/AbstractScript.java

```diff
@@ -59,10 +60,16 @@ private void processEvents(List<NodeEvent> events, String nodeId, List<ScriptLis
     }
 
     public void addGenericScript(GenericScript genericScript) {
+        if (genericScript.hasDeepDependency(this) || getId().equals(genericScript.getId())) {
```

Comment:
I would move second check first because it is the fastest one? (The other one could be time consuming)